src/dmqnode/storage/Ring.d

```diff
@@ -315,18 +315,21 @@ public class RingNode : StorageChannels
 
         override public typeof(this) pop ( ref char[] value )
         {
-            void[] allocValue ( size_t n )
+            scope allocValue = delegate void[] ( size_t n )
             {
+                if (value.length < n)
+                    enableStomping(value);
+
                 value.length = n;
                 return value;
-            }
+            };
 
             if (void[] item = this.queue.pop())
             {
                  allocValue(item.length)[] = item[];
                  this.outer.dmqnode.record_action_counters.increment("popped", value.length);
             }
-            else if (this.overflow.pop(&allocValue))
+            else if (this.overflow.pop(allocValue))
```

Comment:
Why is this necessary ? `d1to2fix` should insert `scope` [here](https://github.com/sociomantic-tsunami/dmqnode/blob/7696571f435275e939730daace651ad4ee1087e7/src/dmqnode/storage/engine/DiskOverflow.d#L731) 🤔